pandas/core/describe.py

```diff
@@ -376,3 +380,8 @@ def refine_percentiles(percentiles: Optional[Sequence[float]]) -> Sequence[float
         raise ValueError("percentiles cannot contain duplicates")
 
     return unique_pcts
+
+
+def validate_frame(frame: "DataFrame"):
```

Comment:
a reason not to do this in the constructor? e.g. L148

Reply:
I though of simplifying the constructor.
Ideally I would create a method ``_initialize_frame``, which will perform the validation under the hood and return the frame.

```
self.frame = self._initialize_frame(frame)
```

But that will be a static method, which you prefer not to use.

If you consider that this extraction is unnecessary, then I can definitely inline it into the constructor.

Reply:
Moved into the constructor.